zebedee-cms/src/main/java/com/github/onsdigital/zebedee/session/service/NewSessionsServiceImpl.java

```diff
@@ -68,72 +67,70 @@ public Session get(HttpServletRequest id) throws IOException {
      * Get a {@link Session} by it's ID.
      *
      * @param id the ID of the session to get,
-     * @return the {@link Session} instance if it exists and is not expired.
+     * @return the {@link Session} instance if it exists and is not expired. If the id is null or empty will return null.
      * @throws IOException for any problems getting the session.
      */
     @Override
     public Session get(String id) throws IOException {
-        com.session.service.Session clientSession = client.getSessionByID(id);
+        if (StringUtils.isEmpty(id)) {
+            return null;
```

Comment:
Consider throwing an exception here instead of returning null. That way you can add detail to the exception message telling the caller exactly what the issue is. We should probably add test coverage for these cases as well.

Reply:
Hi Carl. Thanks for the comment. Their is a reason this needs to return a null.
When adding a collection the keychain checks anyone with access to the collection. If that user hasn't logged in it has no session so returns an error when it doesn't really need to. Not sure if there is a bit more graceful of a way around it at the minute.

Reply:
@CarlHembrough we did this initially but we broke backwards compatibility. The current / existing functionality expects this to be `null` in some cases. The ripple effect of doing this "properly" is pretty wide spread so we are going to keep this as null.